Fix-Raiden-Boss 2.0 (for all user )/RaidenBossFix2.py

```diff
@@ -24,85 +26,541 @@
+        for i in range(0,len(blendData),32):
+            blendweights = [struct.unpack("<f", blendData[i+4*j:i+4*(j+1)])[0] for j in range(4)]
+            blendindices = [struct.unpack("<I", blendData[i+16+4*j:i+16+4*(j+1)])[0] for j in range(4)]
+            outputweights = bytearray()
+            outputindices = bytearray()
+            for weight, index in zip(blendweights, blendindices):
+                if weight != 0 and index <= MaxVGIndex:
+                    index = int(VGRemap[str(index)])
+                outputweights += struct.pack("<f", weight)
+                outputindices += struct.pack("<I", index)
+            result += outputweights
+            result += outputindices
+        FixedBlendName = f"{blendFile.split('Blend.buf')[0]}RemapBlend.buf"
+        with open(FixedBlendName, "wb") as f:
+            f.write(result)
+        self._logger.log('Blend file correction done')
+        return FixedBlendName
+
+    # fix each individual mod containing the assets
+    def fixBaseMod(self, path: str = DefaultPath, iniFile: Optional[str] = None, blendFile: Optional[str] = None, files: Optional[List[str]] = None) -> RemapBlendModel:
+        if (iniFile is None and blendFile is None):
+            iniFile, blendFile = ModFileService.getBaseModFiles(path = path, files = files)
+        elif (iniFile is None):
+            iniFile = ModFileService.getIni(path = path, files = files)
+        elif (blendFile is None):
+            blendFile = ModFileService.getBlend(path = path, files = files)
+
+        iniLines = self._iniFileService.getFileLines(iniFile)
+        textureOverideBlendDicts = self._iniFileService.getTextureOverideBlendDicts(iniLines)
+        draw = self._iniFileService.getBlendDrawValue(textureOverideBlendDicts[0])
+
+        fixedBlendName = self._blendCorrection(blendFile)
+        fixedBlendName = os.path.basename(fixedBlendName).split('.')[0]
+        
+        remapBlendModel = RemapBlendModel(fixedBlendName, draw, path)
+
+        # writing the fixed file
+        self._logger.log("Making the fixed ini file")
+        self._iniFileService.fixBase(iniFile, remapBlendModel, logger = self._logger)
+
+        return remapBlendModel
+
+
+    def _fix(self):
+        modFolder = os.path.basename(os.getcwd())
+        self._loggerBasePrefix = modFolder
+        self._logger.prefix = self._loggerBasePrefix
+
+        files, dirs = FileService.getFilesAndDirs()
+        topIniFile = ModFileService.getIni(files = files)
+
+        if (ModFileService.isMerged(topIniFile)):
+            self._logger.log(f"Reading {MergedFile} file for individual mods to modify")
+
+            # read the merged.ini for mod folders
+            iniLines = self._iniFileService.getFileLines(topIniFile)
+            resourceBlendDicts = self._iniFileService.getResourceBlendDicts(iniLines)
+
+            modFolders = self._iniFileService.getResourceBlendFolderPaths(resourceBlendDicts)
+            remapBlendModelsDict = {}
```

Comment:
Sacrifice a bit of space for speed by searching duplicate folders using a Hashmap/Dictionary